sdk/ai/Azure.AI.Projects/README.md

```diff
@@ -0,0 +1,351 @@
+`ToolOutput` instances. For convenience, a basic example is extracted here:
+
+```C# Snippet:FunctionsHandleFunctionCalls
+ToolOutput GetResolvedToolOutput(RequiredToolCall toolCall)
+{
+    if (toolCall is RequiredFunctionToolCall functionToolCall)
+    {
+        if (functionToolCall.Name == getUserFavoriteCityTool.Name)
+        {
+            return new ToolOutput(toolCall, GetUserFavoriteCity());
+        }
+        using JsonDocument argumentsJson = JsonDocument.Parse(functionToolCall.Arguments);
+        if (functionToolCall.Name == getCityNicknameTool.Name)
+        {
+            string locationArgument = argumentsJson.RootElement.GetProperty("location").GetString();
+            return new ToolOutput(toolCall, GetCityNickname(locationArgument));
+        }
+        if (functionToolCall.Name == getCurrentWeatherAtLocationTool.Name)
+        {
+            string locationArgument = argumentsJson.RootElement.GetProperty("location").GetString();
+            if (argumentsJson.RootElement.TryGetProperty("unit", out JsonElement unitElement))
+            {
+                string unitArgument = unitElement.GetString();
+                return new ToolOutput(toolCall, GetWeatherAtLocation(locationArgument, unitArgument));
+            }
+            return new ToolOutput(toolCall, GetWeatherAtLocation(locationArgument));
+        }
+    }
+    return null;
+}
+```
+
+To handle user input like "what's the weather like right now in my favorite city?", polling the response for completion
+should be supplemented by a `RunStatus` check for `RequiresAction` or, in this case, the presence of the
+`RequiredAction` property on the run. Then, the collection of `ToolOutputSubmissions` should be submitted to the
+run via the `SubmitRunToolOutputs` method so that the run can continue:
+
+```C# Snippet:FunctionsHandlePollingWithRequiredAction
+do
+{
+    await Task.Delay(TimeSpan.FromMilliseconds(500));
+    runResponse = await client.GetRunAsync(thread.Id, runResponse.Value.Id);
+
+    if (runResponse.Value.Status == RunStatus.RequiresAction
+        && runResponse.Value.RequiredAction is SubmitToolOutputsAction submitToolOutputsAction)
+    {
+        List<ToolOutput> toolOutputs = new();
+        foreach (RequiredToolCall toolCall in submitToolOutputsAction.ToolCalls)
+        {
+            toolOutputs.Add(GetResolvedToolOutput(toolCall));
+        }
+        runResponse = await client.SubmitToolOutputsToRunAsync(runResponse.Value, toolOutputs);
+    }
+}
+while (runResponse.Value.Status == RunStatus.Queued
+    || runResponse.Value.Status == RunStatus.InProgress);
+```
+
+## Troubleshooting
+
+When you interact with Azure AI Projects using the .NET SDK, errors returned by the service correspond to the same HTTP status codes returned for REST API requests.
+
+For example, if you try to create a client using an endpoint that doesn't match your Azure AI Resource endpoint, a `404` error is returned, indicating `Resource Not Found`.
+
```

Comment:
Do you know how to turn on SDK logs? That's a very useful tool for troubleshooting, and good to show here as part of troubleshooting section (or provide a link)

Reply:
Sure, I'll add https://learn.microsoft.com/en-us/dotnet/azure/sdk/logging link